plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1266,6 +1265,17 @@ void producer_plugin::plugin_startup()
             chain.init_thread_local_data();
          });
 
+      // Do a test run to make sure all threads have started
+      std::vector<std::future<bool>> fut;
+      for (auto i = 0; i < my->_ro_thread_pool_size; ++i ) {
+         fut.emplace_back( post_async_task( my->_ro_thread_pool.get_executor(), [] () {
```

Comment:
These do not all necessarily execute on diff threads. I would say it is unlikely that they would. Instead, I would use an `std::atomic<uint32_t>` that was incremented after `chain.init_thread_local_data();` above. Then either wait on condition variable that is signaled after the uint32_t increment. Or just a simple while loop waiting on the uint32_t to equal `my->_ro_thread_pool_size`. The condition variable needs a mutex which seems like a lot for this. Too bad we don't have c++20 yet as this could be done trivially with a `std::latch` or `std::atomic<>::wait`. For now I would say the simple loop with a 1ms sleep is likely the best with a comment that this can switch to a `std::latch` once C++20 is available.

Reply:
Thanks for your detailed suggestion. Will use a simple loop since this only executes once at startup.